### PR TITLE
Compatibility with php 8.4

### PR DESCRIPTION
**Pull request by srjlewis**
@Geolim4 hope this helps

Also could you do a release of PHPSocialNetwork/couchbasev4-extension 😊

Deprecate implicitly nullable parameter types
https://wiki.php.net/rfc/deprecate-implicitly-nullable-types

### CODE REVIEW

**Comment by Geolim4** [comment]
Thanks, I take a look tonight !

**Comment by Geolim4** [comment]
I think I'm gonna throw travis Ci as it become more and more unstable over the years...

**Comment by srjlewis** [comment]
Yep I dont blame you, I do think git actions dose a better job these days.

**Comment by srjlewis** [comment]
@Geolim4 

Could I give you a little nudge as a reminder on this pull 😁

Also do you mind if I push a release out for PHPSocialNetwork/couchbasev4-extension or would you prefur to do it?

**Comment by Geolim4** [comment]
Can you add a commit to remove travis.yml please ?
As they don't support it, I have to get rid of them to keep only Github Actions.

I'll merge it tonight !

**Comment by srjlewis** [comment]
Yes I have removed it for you.

**Comment by srjlewis** [comment]
@Geolim4 It also looks like Scrutinizer is have trouble, it cant even compile PHP 8.0 and throwing a error.

**Comment by srjlewis** [comment]
@Geolim4 I know you are a busy man, just a reminder nudge. 😊

**Comment by Geolim4** [comment]
Can you take a look at why the tests fails on GetAllItems on 8.4 ?
I don't know if its related to this PR ?

**Comment by srjlewis** [comment]
Hi @Geolim4 

TLDR: not related to this PR

The test that failed was on PHP 8.3.13

I was having a look yesterday when I see the tests fail. It looks like a redis, some thing has changed in the setup.

When I first subbmitted the PR all tests passed on my fork [Attempt 1](https://github.com/srjlewis/phpfastcache/actions/runs/11496639500/attempts/1)
When I see the tests fail yesterday I reran the same tests on my fork and it failed [Attempt 2](https://github.com/srjlewis/phpfastcache/actions/runs/11496639500)

I have been able to get the tests to pass byr the following commit on a test branch https://github.com/srjlewis/phpfastcache/commit/61d2acf3ddc7f8704c333b6c0a2db046b74022fa by removing redis from [GetAllItems.test.php](https://github.com/srjlewis/phpfastcache/blob/61d2acf3ddc7f8704c333b6c0a2db046b74022fa/tests/cases/GetAllItems.test.php)

I have also noticed that the tests are still running against redis 5, would it be a better idea to move redis to an extention and then also test against redis 6 and 7, just a idea so dont shot me 🤪

**Comment by knyjoh** [comment]
Hello, I'm adding a small nudge to this one. Would it be possible to get this one over the line to make it PHP 8.4 compatible?

**Comment by Geolim4** [comment]
Thank you @srjlewis, was merged in another PR that someone reasked me today.
I'll be more active these next months !
